### PR TITLE
Add function  to get email from specific user

### PR DESCRIPTION
**Pull request by mojtaba-fs**
https://redmine.deriv.cloud/issues/52874#affiliate/avoid_self_tagging

### CODE REVIEW

File: lib/WebService/MyAffiliates.pm [hunk 1]
```diff
@@ -165,6 +165,15 @@ sub get_affiliate_id_from_token {
     return $token_info->{TOKEN}->{USER_ID};
 }
 
+sub get_affiliate_email_from_token {
+    my ($self, $token) = @_;
+
+    $token or croak 'Must pass a token to get_affiliate_id_from_token.';
```
Comment (sasikala-binary) [inline lib/WebService/MyAffiliates.pm:171]:
```suggestion
    $token or croak 'Must pass a token to get_affiliate_email_from_token.';
```

File: lib/WebService/MyAffiliates.pm [hunk 2]
```diff
@@ -165,6 +165,15 @@ sub get_affiliate_id_from_token {
     return $token_info->{TOKEN}->{USER_ID};
 }
 
+sub get_affiliate_email_from_token {
```
Comment (sasikala-binary) [inline lib/WebService/MyAffiliates.pm:168]: Please add test for this subroutine

Comment (shantanu-deriv) [reply lib/WebService/MyAffiliates.pm:168]: Also let us add POD for newly created subs.
https://wikijs.deriv.cloud/en/Backend/guides/perl/perl-style-guide#subroutine-comments

Comment (mehdi-fs) [reply lib/WebService/MyAffiliates.pm:168]: Ah I put reviews on cpan :( , please address regentmarkets/cpan#2785 as well.

**Review by qiangliu-binary — APPROVED** [review]

File: t/decode_token.t [hunk 3]
```diff
@@ -25,6 +25,9 @@ is(ref $token_info->{'TOKEN'}, 'ARRAY', 'Got data back about two tokens, so the
 my $affiliate_id = $aff->get_affiliate_id_from_token('jGZUKO3JWgyVAv0U_Fv2nVOqZLGcUW5p');
 is($affiliate_id, 6, 'Token has affiliate id 6');
 
+my $affiliate_details = $aff->get_affiliate_details('jGZUKO3JWgyVAv0U_Fv2nVOqZLGcUW5p');
```
Comment (qiangliu-binary) [inline t/decode_token.t:28]: please add more tests for like 
1. the token is invalid
2. token is empty
3. maybe more senario

**Pull request merged**